pkg/cloud/azure/provider.go

```diff
@@ -916,6 +921,30 @@ func (az *Azure) DownloadPricingData() error {
 		}()
 	}
 
+	envResourceGroupName := env.GetAzureResourceGroupName()
+
+	envManagedClustersName := env.GetAzureClusterName()
+
+	err = az.refreshClusterManagementPricing(config, envResourceGroupName, envManagedClustersName)
```

Comment:
Can we wrap all of this in an if statement so that we are not running it if these env variables are not set.

Reply:
Done